### ISSUE TITLE
Support use of projects as instructions to be made in other projects. 

### ISSUE THREAD

**Issue by quachtina96** [quachtina96/scratch-vui#13]
e.g. When I make a project called "say my name in cat language" and the instructions are "play the meow sound", Can I make a new project called "i really think you are a cat", and have its instruction be "say my name in cat language 5 times".

This also raises another question. Beyond using the should projects be able to be recursive? Thinking about recursive functions...useful recursive functions usually take inputs or operate on some variable that is maintained and updated outside of the recursive helper function. Currently, the projects do not provide a way for the user to provide input.... but that would be doable!

**Comment by quachtina96** [comment]
This also requires that projects names cannot overlap with Scratch commands.

**Comment by quachtina96** [comment]
What should happen when a user modifies a project (that is used as an instruction in other projects)?
- all uses of that project as an instruction will also be updated (but only after warning the user)
What should happen when a user deletes a project (that is used as an instruction in other projects)?- First, warn the user that a delete will cause the following: all uses of that project as an instruction will do nothing when that instruction occurs OR just remove that instruction. If user confirms, just do it :)

**Comment by quachtina96** [comment]
This could get complicated when considering "play step" where playing a step refers to executing an entire project. 

Maybe would could happen is that the project could get played, but it would still happen while you're in the "inside/editing project state".

**Comment by quachtina96** [comment]
might be more of a future work thing.
